egglog-reports/src/lib.rs

```diff
@@ -0,0 +1,262 @@
+use clap::clap_derive::ValueEnum;
+use rustc_hash::FxHasher;
+use serde::Serialize;
+use std::{
+    fmt::{Display, Formatter},
+    hash::BuildHasherDefault,
+};
+use web_time::Duration;
+
+pub(crate) type HashMap<K, V> = hashbrown::HashMap<K, V, BuildHasherDefault<FxHasher>>;
+// pub(crate) type HashSet<T> = hashbrown::HashSet<T, BuildHasherDefault<FxHasher>>;
+pub(crate) type IndexSet<T> = indexmap::IndexSet<T, BuildHasherDefault<FxHasher>>;
+// pub(crate) type IndexMap<K, V> = indexmap::IndexMap<K, V, BuildHasherDefault<FxHasher>>;
+
+#[derive(ValueEnum, Default, Serialize, Debug, Clone, Copy)]
+pub enum ReportLevel {
+    #[default]
+    SizeOnly,
+    WithPlan,
+    StageInfo,
+}
```

Comment:
I was also thinking it might be nice to add an option to not save any reporting info, to make it more performant. Just a fast pass that skips any of this logic that could be optionally enabled. However, this could be added in a follow up PR.